runtime/doc/treesitter.txt

```diff
@@ -1167,10 +1167,17 @@ LanguageTree:named_node_for_range({range}, {opts})
     Return: ~
         |TSNode| | nil Found node
 
-LanguageTree:parse()                                    *LanguageTree:parse()*
+LanguageTree:parse({opts})                              *LanguageTree:parse()*
     Parses all defined regions using a treesitter parser for the language this
-    tree represents. This will run the injection query for this language to
-    determine if any child languages should be created.
+    tree represents.
+
+    If {range} is provided and isn't `false`, will query and parse injections.
+
+    Parameters: ~
+      • {opts}  TSParseOpts|nil Optional arguments:
+                • range boolean|Range4: Only parse this range in the buffer.
+                  This only affects trees which have an explicit range defined
+                  (injections) and does not affect the root tree.
```

Comment:
This is slightly ambiguous: Does it parse the range, or just _injections_ in the range?

(And the second sentence probably fits better in the preceding paragraph.)

Reply:
The languagetree has "regions". A region can be empty (undefined) which means parse the whole ~buffer~ source. `range` does not affect these undefined regions.

So if you create a language tree and run `set_included_regions`, you've now defined regions to be parsed and thus `range` will be applied.

Injections are always defined regions, so `range` always applies.

I'll try and tidy this up. It was rushed.